src/backend/storage/smgr/md.c

```diff
@@ -516,7 +516,18 @@ mdunlinkfork(RelFileNodeBackend rnode, ForkNumber forkNum, bool isRedo, char rel
 	 */
 	if (isRedo || forkNum != MAIN_FORKNUM || RelFileNodeBackendIsTemp(rnode))
 	{
+/*
+ * GPDB: Temp tables use shared buffers in Greenplum. As a result, simple unlink
+ * is not enough to return disk space to the OS immediately, because the files
+ * of a temp relation can still be opened by the bg writer process. To fix this
+ * problem, in GPDB we skip the check if the relation is temp or not, so files
+ * of the temp relation are also truncated.
+ */
+#if 0
 		if (!RelFileNodeBackendIsTemp(rnode))
+#else
+		if (true)
```

Comment:
Why is there an #else block here, but in the next similar change it is no longer there?

Reply:
Because it has 
```
		else
			ret = 0;
```
Of course we can cut off the 'else' branch, but I think that current change gives less delta.